relay/relay.go

```diff
@@ -9,57 +9,50 @@ import (
 	"net"
 	"time"
 
-	"github.com/offchainlabs/nitro/arbstate"
 	"github.com/offchainlabs/nitro/arbutil"
 	"github.com/offchainlabs/nitro/broadcastclient"
 	"github.com/offchainlabs/nitro/broadcaster"
 	"github.com/offchainlabs/nitro/util/stopwaiter"
-	"github.com/offchainlabs/nitro/wsbroadcastserver"
 )
 
 type Relay struct {
 	stopwaiter.StopWaiter
 	broadcastClients            []*broadcastclient.BroadcastClient
 	broadcaster                 *broadcaster.Broadcaster
 	confirmedSequenceNumberChan chan arbutil.MessageIndex
-	messageChan                 chan broadcastFeedMessage
-}
-
-type broadcastFeedMessage struct {
-	message        arbstate.MessageWithMetadata
-	sequenceNumber arbutil.MessageIndex
+	messageChan                 chan broadcaster.BroadcastFeedMessage
 }
 
 type RelayMessageQueue struct {
-	queue chan broadcastFeedMessage
+	queue chan broadcaster.BroadcastFeedMessage
 }
 
-func (q *RelayMessageQueue) AddBroadcastMessages(pos arbutil.MessageIndex, messages []arbstate.MessageWithMetadata) error {
-	for i, message := range messages {
-		q.queue <- broadcastFeedMessage{
-			sequenceNumber: pos + arbutil.MessageIndex(i),
-			message:        message,
-		}
+func (q *RelayMessageQueue) AddBroadcastMessages(feedMessages []*broadcaster.BroadcastFeedMessage) error {
+	for _, feedMessage := range feedMessages {
+		q.queue <- *feedMessage
 	}
 
 	return nil
 }
 
-func NewRelay(serverConf wsbroadcastserver.BroadcasterConfig, clientConf broadcastclient.BroadcastClientConfig, chainId uint64, feedErrChan chan error) *Relay {
+func NewRelay(feedConfig broadcastclient.FeedConfig, chainId uint64, feedErrChan chan error) *Relay {
 	var broadcastClients []*broadcastclient.BroadcastClient
 
-	q := RelayMessageQueue{make(chan broadcastFeedMessage, 100)}
+	q := RelayMessageQueue{make(chan broadcaster.BroadcastFeedMessage, 100)}
 
 	confirmedSequenceNumberListener := make(chan arbutil.MessageIndex, 10)
 
-	for _, address := range clientConf.URLs {
-		client := broadcastclient.NewBroadcastClient(address, chainId, 0, clientConf.Timeout, &q, feedErrChan)
+	for _, address := range feedConfig.Input.URLs {
+		client := broadcastclient.NewBroadcastClient(feedConfig.Input, address, chainId, 0, &q, feedErrChan, nil)
```

Comment:
It's probably not worth adding it, but just noting that the relays won't verify signatures. That's probably fine because the end clients will, and it'd be hard to add as the relays don't have an L1 connection to know which sequencer keys are valid.